test/extensions/http/header_validators/envoy_default/http_common_validation_test.cc

```diff
@@ -0,0 +1,87 @@
+#include "envoy/http/header_validator_errors.h"
+
+#include "source/extensions/http/header_validators/envoy_default/http1_header_validator.h"
+#include "source/extensions/http/header_validators/envoy_default/http2_header_validator.h"
+
+#include "test/extensions/http/header_validators/envoy_default/header_validator_test.h"
+#include "test/test_common/test_runtime.h"
+#include "test/test_common/utility.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace Http {
+namespace HeaderValidators {
+namespace EnvoyDefault {
+namespace {
+
+using ::Envoy::Http::Protocol;
+using ::Envoy::Http::TestRequestHeaderMapImpl;
+
+enum class Protocols {
```

Comment:
should just reuse same enum class in envoy/http/codec.h, the CodecType ?
 
Also  having an `enum Protocol` in scope from the using above and an `enum Protocols` defined here is very easy to misread; this would avoid

Reply:
Done